sdks/python/apache_beam/io/gcp/bigquery_io_read_pipeline.py

```diff
@@ -14,66 +14,135 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-"""A Dataflow job that counts the number of rows in a BQ table.
-
-   Can be configured to simulate slow reading for a given number of rows.
+"""
+A pipeline that reads data from a BigQuery table and counts the number of
+rows.
+
+Besides of the standard options, there are options with special meaning:
+* input_dataset - BQ dataset id.
+* input_table - BQ table id.
+The table will be created and populated with data from Synthetic Source if it
+does not exist.
+* input_options - options for Synthetic Source:
+num_records - number of rows to be inserted,
+value_size - the length of a single row,
+key_size - required option, but its value has no meaning,
+* num_slow - an integer in range [0,100] used to customize slow reading
+simulation.
+
+Example test run on DataflowRunner:
+
+python setup.py nosetests \
+    --test-pipeline-options="
+    --runner=TestDataflowRunner
+    --project=...
+    --staging_location=gs://...
+    --temp_location=gs://...
+    --sdk_location=.../dist/apache-beam-x.x.x.dev0.tar.gz
+    --input_dataset=...
+    --input_table=...
+    --input_options='{
+    \"num_records\": 1024,
+    \"key_size\": 1,
+    \"value_size\": 1024,
+    }'" \
+    --tests apache_beam.io.gcp.tests.bigquery_io_read_pipeline
 """
 
 from __future__ import absolute_import
 
-import argparse
+import base64
 import logging
 import random
 import time
-
-import apache_beam as beam
-from apache_beam.options.pipeline_options import PipelineOptions
+import unittest
+
+from apache_beam import DoFn
+from apache_beam import Map
+from apache_beam import ParDo
+from apache_beam.io import BigQueryDisposition
+from apache_beam.io import BigQuerySource
+from apache_beam.io import Read
+from apache_beam.io import WriteToBigQuery
+from apache_beam.io.gcp.bigquery_tools import BigQueryWrapper
+from apache_beam.io.gcp.bigquery_tools import parse_table_schema_from_json
+from apache_beam.testing.load_tests.load_test import LoadTest
+from apache_beam.testing.synthetic_pipeline import SyntheticSource
 from apache_beam.testing.test_pipeline import TestPipeline
 from apache_beam.testing.util import assert_that
 from apache_beam.testing.util import equal_to
-
-
-class RowToStringWithSlowDown(beam.DoFn):
```

Comment:
The existing `_io_read_it_test.py` assumed that input data are already created. I decided that it could use my new pipeline which covers the possibility that no data exist. It seems it doesn't have any side effects, since all Jenkins jobs passed.